activesupport/test/dependencies_test.rb

```diff
@@ -185,6 +185,44 @@ def test_require_dependency_interaction_with_autoloading
     end
   end
 
+  # Regression see https://github.com/rails/rails/issues/31694
+  def test_included_constant_that_changes_to_have_exception_then_back_does_not_loop_forever
+    path = File.join(__dir__, "autoloading_fixtures/another_constant/reload_error.rb")
+    original_text = File.read(path)
+    error_text = original_text.sub("# no_such_method_as_this", "no_such_method_as_this")
+
+    with_autoloading_fixtures do
+      # This constant is accessable:
+      assert ConstantReloadError
+      assert AnotherConstant::ReloadError
+
+      Timeout.timeout(0.1) do
+        # Remove the constant, as if Rails development middleware is reloading changed files:
+        ActiveSupport::Dependencies.remove_unloadable_constants!
+        refute defined?(AnotherConstant::ReloadError)
+      end
+
+      # Change the file so that there is an error in it:
+      File.write(path, error_text)
```

Comment:
Rewriting the original/source fixture file seems needlessly fragile.

Let's construct a tmpdir for this test, which we can safely scribble on.

Reply:
Thanks for the suggestion, I never used `tmpdir` before! I pushed a refactor in 7170e91b33, is that what you had in mind?